apps/minifront/src/state/ibc-in/index.tsx

```diff
@@ -207,6 +208,24 @@ async function execute(
   const assetMetadata = augmentToAsset(coin.raw.denom, selectedChain.chainName);
 
   const transferToken = fromDisplayAmount(assetMetadata, coin.displayDenom, amount);
+
+  const { address: t_addr, encoding: encoding } = await penumbra
+    .service(ViewService)
+    .transparentAddress(new TransparentAddressRequest({}));
+  if (!t_addr) {
+    throw new Error('Error with generating IBC transparent address');
+  }
+
+  // Temporary: detect USDC Noble inbound transfers, and use transparent (t-addr) encoding
+  // to ensure bech32m encoding compatibility.
+  if (
+    transferToken.denom.includes('uusdc') &&
+    (selectedChain.chainId === 'noble-1' || selectedChain.chainId === 'grand-1')
+  ) {
+    // Set the reciever address to the t-addr encoding.
+    penumbraAddress = encoding;
+  }
```

Comment:
Shouldn't this also be setting the flag in the withdrawal message that tells the Penumbra chain to encode the address as a tpenumbra1... address? Otherwise I think it will be the `penumbra1...` encoding of the user's t-addr.

Reply:
Oh, no, because this is ibc-IN